common/wallet_tx.c

```diff
@@ -39,6 +43,63 @@ struct command_result *param_wtx(struct command *cmd,
         return NULL;
 }
 
+struct command_result *param_utxos(struct command *cmd,
+				 const char *name,
+				 const char *buffer,
+				 const jsmntok_t *tok,
+				 const struct utxo ***utxos)
+{
+	size_t i;
+	const jsmntok_t *curr;
+	struct bitcoin_txid **txids = tal_arr(cmd, struct bitcoin_txid*, 0);
+	unsigned int **outnums = tal_arr(cmd, unsigned int*, 0);
+
+	json_for_each_arr(i, curr, tok) {
+		/* A txid is always 32 bytes (64 hex characters) long */
+        const jsmntok_t txid_tok = {JSMN_STRING, curr->start,
+									curr->start + 64, curr->size};
+		/* Pass the `:` separator */
+		const jsmntok_t outnum_tok = {JSMN_STRING, txid_tok.end + 1,
+									curr->end, curr->size};
```

Comment:
This code can read past the end of malformed input. 

Prefer something like this:

```
static bool split_tok(const char *buffer, const jsmntok_t *t, char split, jsmntok_t *a, jsmntok_t *b)
{
        const char *p = memchr(buffer + t->start, split, buffer + t->end);

        if (!p)
                 return false;

        *a = *b = *t;
        a->end = p - 1 - buffer;
        b->start = p + 1 - buffer;
        return true;
}
```

Reply:
That was what I had at first, but I need `const` tokens and didnt want to do multiple cast. Is it preferable to cast ?